client/daemon.go

```diff
@@ -389,3 +413,17 @@ func (k *KeysFile) Overwrite(ks []string) error {
 	}
 	return ioutil.WriteFile(k.fn, buffer.Bytes(), 0666)
 }
+
+func identifyLockHolders(filename string) (string, error) {
+	if runtime.GOOS != "linux" {
+		return "", errors.New("error identifying lock holder: works only on linux")
+	}
+
+	cmd := exec.Command("lsof", filename)
+	stdout, err := cmd.Output()
+	if err != nil {
+		return "", fmt.Errorf("error identifying lock holder: %w", err)
+	}
```

Comment:
seems kinda hacky, is there a native / library we can use?

Reply:
Unfortunately wasn't able to find anything suitable, and implementing all of the kernel APIs (and parsing proc files and so on) would require a lot of CGo regardless.

Reply:
If the `lsof` call fails we just ignore it, so calling it here should be safe.